### PR TITLE
Use "Microsoft.Web/geoRegions" API filter unsupported region

### PR DESCRIPTION
**Pull request by Flanker32**
- Use "Microsoft.Web/geoRegions" API filter unsupported region
- Move pricing tier UI component above region as region is affected by the selection of pricing tier.

> Configuration Validation will be committed in next pr

### CODE REVIEW

File: Utils/azuretools-core/src/com/microsoft/azuretools/core/mvp/model/webapp/AzureWebAppMvpModel.java [hunk 1]
```diff
@@ -696,6 +699,18 @@ public void clearWebAppsCache() {
         subscriptionIdToWebApps.clear();
     }
 
+    public List<Region> getAvailableRegions(String subscriptionId, PricingTier pricingTier) throws IOException {
```
Comment (andxu) [inline Utils/azuretools-core/src/com/microsoft/azuretools/core/mvp/model/webapp/AzureWebAppMvpModel.java:702]: handle null pricingTier

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/webapp/webappconfig/slimui/creation/WebAppCreationDialog.java [hunk 2]
```diff
@@ -379,50 +389,53 @@ private void onOK() {
...
-            webAppConfiguration.setWebContainer(cbWebContainer.getSelectedItem() == null ? null :
-                ((WebAppUtils.WebContainerMod) cbWebContainer.getSelectedItem()).getValue());
+            webAppConfiguration.setJdkVersion(getValueFromComboBox(cbJdkVersion, JdkModel::getJavaVersion));
+            webAppConfiguration.setWebContainer(getValueFromComboBox(cbWebContainer,
+                                                                     WebAppUtils.WebContainerMod::getValue));
         }
         webAppConfiguration.setCreatingNew(true);
     }
 
+    private <T, R> R getValueFromComboBox(JComboBox comboBox, Function<T, R> function) {
```
Comment (andxu) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/webapp/webappconfig/slimui/creation/WebAppCreationDialog.java:430]: static

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/webapp/webappconfig/slimui/creation/WebAppCreationDialog.java [hunk 3]
```diff
@@ -379,50 +389,53 @@ private void onOK() {
...
+        return getValueFromComboBox(comboBox, function, null);
+    }
+
+    private <T, R> R getValueFromComboBox(JComboBox comboBox, Function<T, R> function, R defaultValue) {
```
Comment (andxu) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/webapp/webappconfig/slimui/creation/WebAppCreationDialog.java:434]: static

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/webapp/webappconfig/slimui/creation/WebAppCreationDialog.java [hunk 4]
```diff
@@ -210,6 +212,14 @@ public void windowClosing(WindowEvent e) {
         init();
     }
 
+    private void loadRegionsBySku() {
```
Comment (andxu) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/webapp/webappconfig/slimui/creation/WebAppCreationDialog.java:215]: move private method last

File: Utils/azuretools-core/src/com/microsoft/azuretools/core/mvp/model/webapp/AzureWebAppMvpModel.java (hunk 1 above)
Comment (Flanker32) [reply Utils/azuretools-core/src/com/microsoft/azuretools/core/mvp/model/webapp/AzureWebAppMvpModel.java:702]: pricingTier is not required in the rest api, here is the [reference](https://docs.microsoft.com/en-us/rest/api/appservice/listgeoregions/listgeoregions#uri-parameters)

Comment (andxu) [reply Utils/azuretools-core/src/com/microsoft/azuretools/core/mvp/model/webapp/AzureWebAppMvpModel.java:702]: if pricingTier is null, NPE will be thrown: pricingTier.toSkuDescription().tier()

**Pull request merged**